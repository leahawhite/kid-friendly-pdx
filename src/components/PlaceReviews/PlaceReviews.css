.places-item-reviews {
  margin: 0;
  padding: 0;
}

.place-reviews {
  text-align: left;
  margin: 0 10px 0 10px;
}

.place-reviews-header {
  margin-top: 20px;
}

.place-reviews > h2, ul {
  padding: 0;
}

.write-review-btn-container {
  text-align: left;
  margin: 10px;
  padding: 15px 0;
}

.write-review-btn {
  border: 1px solid #0045e6;
  background-color:  #0045e6;
  color: white;
  font-weight: bold;
  font-size: 18px;
  padding: 8px;
  -moz-border-radius: 8px;
  -webkit-border-radius: 8px;
  -o-border-radius: 8px 12px;
  -ms-border-radius: 8px;
  border-radius: 8px;
  margin: 0 auto;
}

.write-review-btn a {
  color: #fff;
}

.write-review-btn span {
  margin-left: 10px;
}

.write-review-btn:hover, .write-review-btn:focus, .write-review-btn:active, .write-review-btn a:hover, .write-review-btn a:focus, .write-review-btn a:active {
  background-color: #002885;
  transition: all 0.3s;
}


