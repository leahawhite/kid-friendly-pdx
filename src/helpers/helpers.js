/*export function numberOfReviews(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  return placereviews.length
}

export function averageReviewRating(place) {
  const placeReviews = reviews.filter(review.place_id == place.id)
  const ratingArray = placeReviews.map(review => review.rating) 
  const ratingsSum = ratingArray.reduce(function(a,b){return a + b})
  const numReviews = this.numberOfReviews(place)
  return ratingsSum / numReviews
}*/

export function readableReviewCount(number) {
  switch(number) {
    case 0:
      return 'no reviews yet'

    case 1:
      return `1 review`

    default:
      return `${number} reviews`
  }
}

export const testPlace = {
  id: 1,
  name: "Atlas Pizza",
  address: "6529 SE Foster Rd",
  city: "Portland",
  state: "OR",
  zipcode: "97206",
  lat: 45.489200,
  lng: -122.595070,
  neighborhood: "SE",
  phone: "503-232-3004",
  website: "http://atlaspizzapdx.com",
  hours: [
    {
      "dayOfWeek": "Monday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Tuesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Wednesday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Thursday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
    {
      "dayOfWeek": "Friday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Saturday",
      "opens": "12:00 pm",
      "closes": "12:00 am",
    },
    {
      "dayOfWeek": "Sunday",
      "opens": "12:00 pm",
      "closes": "11:00 pm",
    },
  ],
  date_created: "2019-06-13",
  category: ["restaurants",],
  descriptors: [ "pizza", "arcade", "beer" ],
  features: { 
    "toys": true, 
    "play area": true, 
    "arcade": true,
    "quick service": true,
    "all ages": true,
    "comfortable seating": true,
    "friendly staff": true,
    "kids menu": false,
    "highchairs/boosters": false,
    "changing station": false,
    "flexible": true,
    "patio/sidewalk": true,
  },
  images: [
    {
      id: 1,
      src: 'https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400',
      title: `Atlas Pizza's 'Jalapeno Popper' pie`,
      place_id: 1,
      user_id: 2,
      review_id: 1,
    },
    {
      id: 2,
      src: 'https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400',
      title: 'A rainbow over Foster Blvd',
      place_id: 1,
      user_id: 1,
      review_id: 2,
    },
  ],
  number_of_reviews: 2,
  average_review_rating: 4.5,
};

export const testReviews = [
  { 
    id: 1,
    star_rating: 5,
    text: "Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, and they call your name for pickup when it's ready. Since it's new, the play area is pretty clean and well-stocked. There's an old church pew there for parents to sit and supervise, but the open floor plan means that you can see the play area from pretty much every table. We've only ordered slices, so we've never had to wait long. We're usually in and out in 30 minutes, which is just about perfect for us.",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 5,
    images: [
      {
        id: 1,
        src: 'https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400',
        title: `Atlas Pizza's play area`,
        place_id: 1,
        user_id: 5,
      },
    ]
  },
  { 
    id: 2,
    star_rating: 4,
    text: "Very easy place to take kids of all ages. Nice toy area with a great variety of toys!",
    date_created: "2019-06-13",
    place_id: 1,
    user_id: 1,
    images: [],
  },
];



