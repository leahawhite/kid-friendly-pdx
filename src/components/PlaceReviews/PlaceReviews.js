import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Review from '../../components/Review/Review'
import './PlaceReviews.css'

export default class PlaceReviews extends Component {
  render() {
  const { place, reviews } = this.props
    
    return (
      <>
        <div className="place-reviews">
          <h2 className="place-reviews-header">Reviews</h2>
          <div className="place-reviews-content">
            <ul className="places-item-reviews">
              {reviews && reviews.length ? 
                reviews.map(review =>
                  <Review key={review.id} review={review} place={place} />)
                : null}
            </ul>
          </div>
        </div>
        <div className="write-review-btn-container">
          <button className="write-review-btn" aria-label="write-review">
            <Link to={{
              pathname: `/places/${place.id}/reviews`,
              state: { place: place }
            }}>
              <FontAwesomeIcon icon="pen" size="sm" />
              <span>Write a review</span>
            </Link>
          </button>
      </div>
    </>
    )
  }
}

PlaceReviews.defaultProps = {
  place: {},
  reviews: []
}

