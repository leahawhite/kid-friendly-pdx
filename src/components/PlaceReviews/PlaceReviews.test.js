import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PlaceReviews from './PlaceReviews';
import { testPlace, testReviews } from '../../helpers/helpers';

describe('PlaceReviews component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <PlaceReviews />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected with props', () => {
    const wrapper = shallow(<PlaceReviews place={testPlace} reviews={testReviews} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})