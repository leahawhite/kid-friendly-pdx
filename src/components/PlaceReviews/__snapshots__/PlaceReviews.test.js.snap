// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceReviews component renders UI as expected with props 1`] = `
<Fragment>
  <div
    className="place-reviews"
  >
    <h2
      className="place-reviews-header"
    >
      Reviews
    </h2>
    <div
      className="place-reviews-content"
    >
      <ul
        className="places-item-reviews"
      >
        <Review
          key="1"
          place={
            Object {
              "address": "6529 SE Foster Rd",
              "average_review_rating": 4.5,
              "category": Array [
                "restaurants",
              ],
              "city": "Portland",
              "date_created": "2019-06-13",
              "descriptors": Array [
                "pizza",
                "arcade",
                "beer",
              ],
              "features": Object {
                "all ages": true,
                "arcade": true,
                "changing station": false,
                "comfortable seating": true,
                "flexible": true,
                "friendly staff": true,
                "highchairs/boosters": false,
                "kids menu": false,
                "patio/sidewalk": true,
                "play area": true,
                "quick service": true,
                "toys": true,
              },
              "hours": Array [
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Monday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Tuesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Wednesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Thursday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Friday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Saturday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Sunday",
                  "opens": "12:00 pm",
                },
              ],
              "id": 1,
              "images": Array [
                Object {
                  "id": 1,
                  "place_id": 1,
                  "review_id": 1,
                  "src": "https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400",
                  "title": "Atlas Pizza's 'Jalapeno Popper' pie",
                  "user_id": 2,
                },
                Object {
                  "id": 2,
                  "place_id": 1,
                  "review_id": 2,
                  "src": "https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400",
                  "title": "A rainbow over Foster Blvd",
                  "user_id": 1,
                },
              ],
              "lat": 45.4892,
              "lng": -122.59507,
              "name": "Atlas Pizza",
              "neighborhood": "SE",
              "number_of_reviews": 2,
              "phone": "503-232-3004",
              "state": "OR",
              "website": "http://atlaspizzapdx.com",
              "zipcode": "97206",
            }
          }
          review={
            Object {
              "date_created": "2019-06-13",
              "id": 1,
              "images": Array [
                Object {
                  "id": 1,
                  "place_id": 1,
                  "src": "https://lh3.googleusercontent.com/QDwk6G6ZCwYzVH-yJHQ824GZ1wKW2XXus8hxxdH-4AAHz-FejC-UVydyE5le1ZvPTIo1cXw1G25qhljwS8dgyIedCCFtIxBonznVQ4LfaPDEZVWmqKS1vFqVUFFNmMmMlKQ5eCpfhNI=w2400",
                  "title": "Atlas Pizza's play area",
                  "user_id": 5,
                },
              ],
              "place_id": 1,
              "star_rating": 5,
              "text": "Atlas is our new favorite spot for a quick, painless dinner with our toddler. It's super casual -- order slices or pies at the counter, and they call your name for pickup when it's ready. Since it's new, the play area is pretty clean and well-stocked. There's an old church pew there for parents to sit and supervise, but the open floor plan means that you can see the play area from pretty much every table. We've only ordered slices, so we've never had to wait long. We're usually in and out in 30 minutes, which is just about perfect for us.",
              "user_id": 5,
            }
          }
          users={Array []}
        />
        <Review
          key="2"
          place={
            Object {
              "address": "6529 SE Foster Rd",
              "average_review_rating": 4.5,
              "category": Array [
                "restaurants",
              ],
              "city": "Portland",
              "date_created": "2019-06-13",
              "descriptors": Array [
                "pizza",
                "arcade",
                "beer",
              ],
              "features": Object {
                "all ages": true,
                "arcade": true,
                "changing station": false,
                "comfortable seating": true,
                "flexible": true,
                "friendly staff": true,
                "highchairs/boosters": false,
                "kids menu": false,
                "patio/sidewalk": true,
                "play area": true,
                "quick service": true,
                "toys": true,
              },
              "hours": Array [
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Monday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Tuesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Wednesday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Thursday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Friday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "12:00 am",
                  "dayOfWeek": "Saturday",
                  "opens": "12:00 pm",
                },
                Object {
                  "closes": "11:00 pm",
                  "dayOfWeek": "Sunday",
                  "opens": "12:00 pm",
                },
              ],
              "id": 1,
              "images": Array [
                Object {
                  "id": 1,
                  "place_id": 1,
                  "review_id": 1,
                  "src": "https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400",
                  "title": "Atlas Pizza's 'Jalapeno Popper' pie",
                  "user_id": 2,
                },
                Object {
                  "id": 2,
                  "place_id": 1,
                  "review_id": 2,
                  "src": "https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400",
                  "title": "A rainbow over Foster Blvd",
                  "user_id": 1,
                },
              ],
              "lat": 45.4892,
              "lng": -122.59507,
              "name": "Atlas Pizza",
              "neighborhood": "SE",
              "number_of_reviews": 2,
              "phone": "503-232-3004",
              "state": "OR",
              "website": "http://atlaspizzapdx.com",
              "zipcode": "97206",
            }
          }
          review={
            Object {
              "date_created": "2019-06-13",
              "id": 2,
              "images": Array [],
              "place_id": 1,
              "star_rating": 4,
              "text": "Very easy place to take kids of all ages. Nice toy area with a great variety of toys!",
              "user_id": 1,
            }
          }
          users={Array []}
        />
      </ul>
    </div>
  </div>
  <div
    className="write-review-btn-container"
  >
    <button
      aria-label="write-review"
      className="write-review-btn"
    >
      <Link
        to={
          Object {
            "pathname": "/places/1/reviews",
            "state": Object {
              "place": Object {
                "address": "6529 SE Foster Rd",
                "average_review_rating": 4.5,
                "category": Array [
                  "restaurants",
                ],
                "city": "Portland",
                "date_created": "2019-06-13",
                "descriptors": Array [
                  "pizza",
                  "arcade",
                  "beer",
                ],
                "features": Object {
                  "all ages": true,
                  "arcade": true,
                  "changing station": false,
                  "comfortable seating": true,
                  "flexible": true,
                  "friendly staff": true,
                  "highchairs/boosters": false,
                  "kids menu": false,
                  "patio/sidewalk": true,
                  "play area": true,
                  "quick service": true,
                  "toys": true,
                },
                "hours": Array [
                  Object {
                    "closes": "11:00 pm",
                    "dayOfWeek": "Monday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "11:00 pm",
                    "dayOfWeek": "Tuesday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "11:00 pm",
                    "dayOfWeek": "Wednesday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "11:00 pm",
                    "dayOfWeek": "Thursday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "12:00 am",
                    "dayOfWeek": "Friday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "12:00 am",
                    "dayOfWeek": "Saturday",
                    "opens": "12:00 pm",
                  },
                  Object {
                    "closes": "11:00 pm",
                    "dayOfWeek": "Sunday",
                    "opens": "12:00 pm",
                  },
                ],
                "id": 1,
                "images": Array [
                  Object {
                    "id": 1,
                    "place_id": 1,
                    "review_id": 1,
                    "src": "https://lh3.googleusercontent.com/ir83SjI1lJpHK4E5vQFwD7LfyE__eKaIvBtmFf1nV5wyfq-z5LVe4u7PuQvQ7ycqsFlGMe5vw3UNRB86G3SjLckIZKHgFSkz0-oQdLAAP2UxlUwq-HqDA116RjcI3iG-lA1ftsYr9S4=w2400",
                    "title": "Atlas Pizza's 'Jalapeno Popper' pie",
                    "user_id": 2,
                  },
                  Object {
                    "id": 2,
                    "place_id": 1,
                    "review_id": 2,
                    "src": "https://lh3.googleusercontent.com/CjUvS4WgCiSRTsUUAryiPcczqAdyaSDhDg1ZTT6OrdgjfdNiTE5Q14pwI-C2m49tSjQHMnBxDgpiZyYZkxEDE3KzI3DWLsT631ROrF1zTuJcBqEXcRN8PP5U3EcG-bmMclE_uKmWh2E=w2400",
                    "title": "A rainbow over Foster Blvd",
                    "user_id": 1,
                  },
                ],
                "lat": 45.4892,
                "lng": -122.59507,
                "name": "Atlas Pizza",
                "neighborhood": "SE",
                "number_of_reviews": 2,
                "phone": "503-232-3004",
                "state": "OR",
                "website": "http://atlaspizzapdx.com",
                "zipcode": "97206",
              },
            },
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="pen"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="sm"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span>
          Write a review
        </span>
      </Link>
    </button>
  </div>
</Fragment>
`;
